distributed/distributed_duckdb_example.ipynb

```diff
@@ -0,0 +1,821 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating scheduler instance\n",
+      "Created instance i-009b530735c2eced3 as dask-2a2e0879-scheduler\n",
+      "Waiting for scheduler to run at 10.5.46.150:8786\n",
+      "Scheduler is running\n",
+      "Creating worker instance\n",
+      "Creating worker instance\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/ssm-user/.pyenv/versions/3.10.12/lib/python3.10/contextlib.py:142: UserWarning: Creating your cluster is taking a surprisingly long time. This is likely due to pending resources. Hang tight! \n",
+      "  next(self.gen)\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Created instance i-0def0e897ddae1b97 as dask-2a2e0879-worker-eae73bf9\n",
+      "Created instance i-015e8fc4dea98a70c as dask-2a2e0879-worker-bf5d0c1f\n"
+     ]
+    }
+   ],
+   "source": [
+    "cluster = EC2Cluster(\n",
+    "    n_workers=2,\n",
+    "    # NGAP AL2023 AMI with default daskdev/dask Docker image already pulled\n",
+    "    ami=\"ami-0db6283751732f2ce\",\n",
+    "    # Default bootstrapping is for Debian-based images only\n",
+    "    bootstrap=False,\n",
+    "    scheduler_instance_type=\"c5.xlarge\",\n",
+    "    worker_instance_type=\"c5.xlarge\",\n",
+    "    # One thread per worker ensures tasks get split across multiple workers\n",
+    "    worker_options={\"nthreads\": 1, \"memory_limit\": \"7GiB\"},\n",
+    "    # debug=True,\n",
+    "    key_name=ssh_key_name,\n",
+    "    region=creds[\"AWS_REGION\"],\n",
+    "    vpc=vpc_id,\n",
+    "    security_groups=[cluster_sg_id],\n",
+    "    extra_bootstrap=bootstrap_script,\n",
+    "    docker_args=docker_args,\n",
+    "    # Default expects public IPs\n",
+    "    use_private_ip=True,\n",
+    "    # TLS doesn't currently work: https://github.com/dask/dask-cloudprovider/issues/249\n",
+    "    security=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div class=\"jp-RenderedHTMLCommon jp-RenderedHTML jp-mod-trusted jp-OutputArea-output\">\n",
+       "    <div style=\"width: 24px; height: 24px; background-color: #e1e1e1; border: 3px solid #9D9D9D; border-radius: 5px; position: absolute;\">\n",
+       "    </div>\n",
+       "    <div style=\"margin-left: 48px;\">\n",
+       "        <h3 style=\"margin-bottom: 0px; margin-top: 0px;\">EC2Cluster</h3>\n",
+       "        <p style=\"color: #9D9D9D; margin-bottom: 0px;\">37ffbd8b</p>\n",
+       "        <table style=\"width: 100%; text-align: left;\">\n",
+       "            <tr>\n",
+       "                <td style=\"text-align: left;\">\n",
+       "                    <strong>Dashboard:</strong> <a href=\"http://10.5.46.150:8787/status\" target=\"_blank\">http://10.5.46.150:8787/status</a>\n",
```

Comment:
Do we want to store and save outputs?  IP address are included.  In Jupyter notebooks you can clear out the output and then save commit the file.

Reply:
At this stage, this work is intended to be a proof of concept rather than a deployed feature. The IP addresses will change every time the cluster is started, and in that sense their inclusion here isn't meaningful. I've preserved outputs in the notebook in order to demonstrate the output to people who don't intend to run the notebook, and to show example output to anyone who tries to run it and may need clues to help debug anything that doesn't run as expected.